### PR TITLE
test(python,rust): Refactor failing test

### PR DESCRIPTION
**Pull request by stinodego**
Related to https://github.com/pola-rs/polars/issues/9122

Changes:
* Refactor the failing test ~and run all OOC tests on one worker~.
* Comment out a Rust test that doesn't compile. See issue https://github.com/pola-rs/polars/issues/9828 for followup.

### CODE REVIEW

File: polars/polars-lazy/polars-plan/src/logical_plan/optimizer/type_coercion/mod.rs [hunk 1]
```diff
@@ -515,27 +515,28 @@ fn early_escape(type_self: &DataType, type_other: &DataType) -> Option<()> {
     }
 }
 
-#[cfg(test)]
```
Comment (ritchie46) [inline polars/polars-lazy/polars-plan/src/logical_plan/optimizer/type_coercion/mod.rs:518]: Can we fix the compilation error?

Comment (stinodego) [reply polars/polars-lazy/polars-plan/src/logical_plan/optimizer/type_coercion/mod.rs:518]: I looked into it but it's not that simple. This test hasn't been run in ages and it shows.

The function being tested `optimize_expr` is no longer a standalone function. And to test it we apparently need to specify expr arena/node and lp_arena/node and I don't really know how that infrastructure works yet.

It's blocking other optimizations to the test suite so I'd rather get rid of it now and address it sometime next month when we can look at it together.

Comment (ritchie46) [reply polars/polars-lazy/polars-plan/src/logical_plan/optimizer/type_coercion/mod.rs:518]: Alright, can you open an issue for that one?

**Comment by ritchie46** [comment]
Out of curiousity? How does that solve the failing test? Was there a race condition?

**Comment by stinodego** [comment]
> Out of curiousity? How does that solve the failing test? Was there a race condition?

Not sure if this will fix it. But generally running multiple tests in one test function makes things a bit more unstable. While this might not help, at least the test will be factored a bit better.

What we really need is some way to control where the disk spill happens, and make sure things are written to a properly controlled temporary directory. Can we configure disk spill somehow?

File: polars/polars-lazy/polars-plan/src/logical_plan/optimizer/type_coercion/mod.rs (hunk 1 above)
Comment (stinodego) [reply polars/polars-lazy/polars-plan/src/logical_plan/optimizer/type_coercion/mod.rs:518]: See #9828

**Comment by ritchie46** [comment]
I think it can be race conditions now that I think of it. The ooc sort/groupby runs a garbage collection job that deletes previous runs. Can we force those tests to not run in parallel?

**Comment by stinodego** [comment]
> I think it can be race conditions now that I think of it. The ooc sort/groupby runs a garbage collection job that deletes previous runs. Can we force those tests to not run in parallel?

Right, so one runner might delete data on disk that a test on another runner relies on.

Sure, we can have all streaming tests on one runner. Let me figure out how to configure that nicely.

**Comment by stinodego** [comment]
This should do it. Place your OOC tests in the `test_ooc.py` module and they should all run on the same worker. Or if your test really must live somewhere else, add the pytest marker `@pytest.mark.xdist_group("ooc")`.

**Comment by stinodego** [comment]
@ritchie46 The tests now run sequentially on a single worker, but it only made things worse 😕 now other tests are failing as well.

Looks like there might be a deeper issue with the garbage collection, or something else that causes subsequent streaming operations to fail?

**Comment by ritchie46** [comment]
Alright, let's leave it for now. I can take a look at this later.

**Comment by stinodego** [comment]
Sure. I'll merge the refactor part, we can address the underlying issues later - I'll open a draft PR for that.

**Pull request merged**